guides/metrics.md

```diff
@@ -0,0 +1,75 @@
+- [SDK Metrics](#SDK-Metrics)
+    - [How to enable metrics](#get-client)
+    - [Metrics Collected](#metrics-collected)
+
+# SDK Metrics
+
+We've integrated a robust metrics solution into the OpenSearch Java client to provide comprehensive insights into its API usage and performance. This includes the collection of vital operational metrics, such as overall throughput, request latency, and error rate. Furthermore, we're capturing more granular details like request and response payload sizes, distinct success and failure rates for operations, and real-time OpenSearch node status to provide a holistic view of client behavior and cluster health.
+
+## How to enable metrics
+
+We should create a MetricOptions instance and set it in the ApacheHttpClient5TransportBuilder when creating the client. Take a look at below code snippet for an example of how to create a client with metrics enabled:
+
+```java
+public class CreateClient {
+  public static OpenSearchClient createClientWithMetrics() throws NoSuchAlgorithmException, KeyStoreException, KeyManagementException {
+    var env = System.getenv();
+    var https = Boolean.parseBoolean(env.getOrDefault("HTTPS", "true"));
+    var hostname = env.getOrDefault("HOST", "localhost");
+    var port = Integer.parseInt(env.getOrDefault("PORT", "9200"));
+    var user = env.getOrDefault("USERNAME", "admin");
+    var pass = env.getOrDefault("PASSWORD", "admin");
+    var metricEnabled = true;
+    double PERCENTILE_99 = 0.99;
+    double PERCENTILE_95 = 0.95;
+    var meterRegistry =  new PrometheusMeterRegistry(PrometheusConfig.DEFAULT);
```

Comment:
maybe use [`SimpleMeterRegistry`](https://javadoc.io/doc/io.micrometer/micrometer-core/latest/io/micrometer/core/instrument/simple/SimpleMeterRegistry.html) for the example since it's not specific to prometheus? it's the simplest example which they also use in [their intro docs](https://docs.micrometer.io/micrometer/reference/concepts/registry.html).